### ISSUE TITLE
aws_s3_bucket_inventory is missing some optional_fields

### ISSUE THREAD

**Issue by vfoucault** [terraform-providers/terraform-provider-aws#7951]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.11
+ provider.aws v1.60.0

### Affected Resource(s)

* aws_s3_bucket_inventory

### Terraform Configuration Files

```hcl
"resource" "aws_s3_bucket_inventory" "testbucket" {
  "bucket" = "testbucket"
  "destination" "bucket" {
    "bucket_arn" = "arn:aws:s3:::desttestbucket"
    "encryption" "sse_s3" {}
    "format" = "Parquet"
  }
  "filter" = {
    "prefix" = "bladibla/"
  }
  "included_object_versions" = "All"
  "name" = "testbucketinventory"
  "optional_fields" = ["ETag", "Size", "StorageClass", "EncryptionStatus", "ObjectLockRetainUntilDate", "ObjectLockMode", "ObjectLockLegalHoldStatus", "ReplicationStatus", "IsMultipartUploaded", "LastModifiedDate"]
  "schedule" = {
    "frequency" = "Daily"
  }
}

```

### Expected Behavior

the bucket inventory should have been created with all the optional fields

### Actual Behavior

Error: aws_s3_bucket_inventory.testbucketinventory: expected optional_fields.4 to be one of [Size LastModifiedDate StorageClass ETag IsMultipartUploaded ReplicationStatus EncryptionStatus], got ObjectLockRetainUntilDate

Error: aws_s3_bucket_inventory.testbucketinventory: expected optional_fields.5 to be one of [Size LastModifiedDate StorageClass ETag IsMultipartUploaded ReplicationStatus EncryptionStatus], got ObjectLockMode

Error: aws_s3_bucket_inventory.testbucketinventory: expected optional_fields.6 to be one of [Size LastModifiedDate StorageClass ETag IsMultipartUploaded ReplicationStatus EncryptionStatus], got ObjectLockLegalHoldStatus

### Steps to Reproduce

1. `terraform plan`

### References

* #0000

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/7158
* https://github.com/terraform-providers/terraform-provider-aws/issues/7159

**Comment by bflad** [comment]
Support for the below additional optional field values has been merged and will release with version 2.3.0 of the Terraform AWS Provider, likely middle of this week. 👍 

* `ObjectLockRetainUntilDate`
* `ObjectLockMode`
* `ObjectLockLegalHoldStatus`

**Comment by bflad** [comment]
This has been released in [version 2.3.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#230-march-21-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
